src/main/scala/ch/epfl/bluebrain/nexus/kg/routes/Routes.scala

```diff
@@ -1,54 +1,183 @@
-                                        aclsOps: AclsOps,
-                                        config: AppConfig): Route = {
-    import indexers._
-    implicit val um           = marshallers.sparqlQueryUnmarshaller
-    implicit val projectCache = cache.project
-    implicit val viewCache    = cache.view
-
-    (handleRejections(RejectionHandling()) & handleExceptions(ExceptionHandling())) {
-      token { implicit optToken =>
-        pathPrefix(config.http.prefix / Segment) { resourceSegment =>
-          project.apply { implicit project =>
-            (acls & caller) { (acl, c) =>
-              resourceSegment match {
-                case "resolvers" => new ResolverRoutes(resources, acl, c).routes
-                case "views"     => new ViewRoutes(resources, acl, c).routes
-                case "schemas"   => new SchemaRoutes(resources, acl, c).routes
-                case "files"     => new FileRoutes(resources, acl, c).routes
-                case "resources" => new ResourceRoutes(resources, acl, c).routes
-                case _           => reject()
-              }
-            }
+  final def wrap(route: Route)(implicit hc: HttpConfig): Route = {
+    val corsSettings = CorsSettings.defaultSettings
+      .withAllowedMethods(List(GET, PUT, POST, DELETE, OPTIONS, HEAD))
+      .withExposedHeaders(List(Location.name))
+    cors(corsSettings) {
+      handleExceptions(exceptionHandler) {
+        handleRejections(rejectionHandler) {
+          uriPrefix(hc.publicUri) {
+            route
           }
         }
       }
     }
   }
+
+  /**
+    * Generates the routes for all the platform resources
+    *
+    * @param resources the resources operations
+    */
+  def apply(resources: Resources[Task])(
+      implicit as: ActorSystem,
+      clients: Clients[Task],
+      cache: Caches[Task],
+      store: FileStore[Task, AkkaIn, AkkaOut],
+      config: AppConfig,
+  ): Route = {
+    import clients._
+    implicit val um: FromEntityUnmarshaller[String] =
+      PredefinedFromEntityUnmarshallers.stringUnmarshaller
+        .forContentTypes(RdfMediaTypes.`application/sparql-query`, MediaTypes.`text/plain`)
+
+    implicit val projectCache: ProjectCache[Task] = cache.project
+    implicit val viewCache: ViewCache[Task]       = cache.view
+
+    val healthStatusGroup = HealthStatusGroup(
+      new CassandraHealthStatus(),
+      new ClusterHealthStatus(Cluster(as)),
+      new IamHealthStatus(clients.iamClient),
+      new AdminHealthStatus(clients.adminClient),
+      new ElasticSearchHealthStatus(clients.elastic),
+      new SparqlHealthStatus(clients.sparql)
+    )
+    val appInfoRoutes = AppInfoRoutes(config.description, healthStatusGroup).routes
+
+    wrap(
+      pathPrefix(config.http.prefix / Segment) { resourceSegment =>
+        extractToken {
+          implicit optToken =>
+            project.apply {
+              implicit project =>
+                (extractCallerAcls & extractCaller) { (acl, c) =>
```

Comment:
I still don't see the point why to make this call to IAM `extractCallerAcls` if we have this information into the cache. 
I think the real fix, which we should do later is to update the cache immediately subscribing to IAM SSE.